src/server/authserver/Server/AuthSession.cpp

```diff
@@ -288,7 +287,7 @@ bool AuthSession::HandleLogonChallenge()
     std::string login((const char*)challenge->I, challenge->I_len);
     TC_LOG_DEBUG("server.authserver", "[AuthChallenge] '%s'", login.c_str());
 
-    if (_queryCallback)
+    if (!_queryProcessor.IsEmpty())
```

Comment:
These checks are not needed anymore since _status guards login sequences and you can only attempt it once per connection